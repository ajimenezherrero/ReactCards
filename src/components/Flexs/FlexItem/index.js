import styled from 'styled-components';

const FlexItem = styled.div`
  flex: ${({ grow, shrink, basis }) => `${grow || 1} ${shrink || ''} ${basis || ''}`};
`;

export default FlexItem;
