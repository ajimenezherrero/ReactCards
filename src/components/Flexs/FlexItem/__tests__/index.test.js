import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import FlexItem from '../index';

describe('<FlexItem />', () => {
  it('renders div  with default value', () => {
    const tree = renderer.create(
      <FlexItem />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders div with flex-grow specified', () => {
    const tree = renderer.create(
      <FlexItem grow={1} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders div with flex-shrink specified', () => {
    const tree = renderer.create(
      <FlexItem shrink={1} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders div with flex-basis specified', () => {
    const tree = renderer.create(
      <FlexItem basis="100%" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
