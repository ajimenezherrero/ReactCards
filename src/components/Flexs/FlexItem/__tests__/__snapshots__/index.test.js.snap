// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlexItem /> renders div  with default value 1`] = `
.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div
  className="c0"
/>
`;

exports[`<FlexItem /> renders div with flex-basis specified 1`] = `
.c0 {
  -webkit-flex: 1 100%;
  -ms-flex: 1 100%;
  flex: 1 100%;
}

<div
  className="c0"
/>
`;

exports[`<FlexItem /> renders div with flex-grow specified 1`] = `
.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div
  className="c0"
/>
`;

exports[`<FlexItem /> renders div with flex-shrink specified 1`] = `
.c0 {
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
}

<div
  className="c0"
/>
`;
